backtesting/backtesting.py

```diff
@@ -1514,3 +1525,68 @@ def plot(self, *, results: pd.Series = None, filename=None, plot_width=None,
             reverse_indicators=reverse_indicators,
             show_legend=show_legend,
             open_browser=open_browser)
+
+    def optimize_skopt(self,
+                       maximize: Union[str, Callable[[pd.Series], float]] = 'SQN',
+                       max_tries: Union[int, float] = 100,
+                       **kwargs) -> Union[pd.Series, Tuple[pd.Series, pd.Series]]:
```

Comment:
No new API methods are planned. I suppose you'll have to find a way to integrate the changes within existing `Backtest.optimize()` API! Maybe simply checking if `param_combos > max_tries`
https://github.com/kernc/backtesting.py/blob/3045b64635fe9e5af51046b7bdf7cdfcbf375c1e/backtesting/backtesting.py#L1237-L1241
and then choose the skopt path.

Then again, `param_combos` can combinatorically explode, and no reason to compute it just to disregard it soon after ... :thinking: 

You're certainly welcome to defer (parts) to a `_optimize_skopt()` _private_ method/coroutine if you see fit.

Reply:
> Then again, `param_combos` can combinatorically explode, and no reason to compute it just to not pay any attention to it later ... :thinking: 

I think we can avoid generating `param_combos` when not needed if we add an additional method argument:
```py
method: Literal['auto', 'full', 'random', 'skopt'] = 'auto'
```

Reply:
Sounds good. I will forge ahead and apply these and more in the next few days...